src/core/net/mdns.cpp

```diff
@@ -1576,6 +1612,36 @@ void Core::HostEntry::AppendNameTo(TxMessage &aTxMessage, Section aSection)
     return;
 }
 
+Error Core::HostEntry::CopyInfoTo(Host &aHost, EntryState &aState) const
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(mAddrRecord.IsPresent(), error = kErrorNotFound);
+
+    aHost.mHostName        = mName.AsCString();
+    aHost.mAddresses       = mAddresses.AsCArray();
+    aHost.mAddressesLength = mAddresses.GetLength();
+    aHost.mTtl             = mAddrRecord.GetTtl();
+    aHost.mInfraIfIndex    = Get<Core>().mInfraIfIndex;
```

Comment:
You never print the mInfraIfIndex ... so I see no point in wasting cycles to get it.

Reply:
CLI is an example app using the APIs. 

The APIs can beused by other (next layer) app code (e.g., when OT stack is integrated into different projects/products).